includes/class-indieauth-remote-authorize.php

```diff
@@ -0,0 +1,180 @@
+					'indieauth_token_endpoint',
+					__( 'Token Endpoint', 'indieauth' ),
+					array( $cls, 'endpoint_field' ),
+					$page,
+					'indieauth',
+					array(
+						'label_for' => 'indieauth_token_endpoint',
+						'class'     => 'widefat',
+						'default'   => 'https://tokens.indieauth.com/token',
+					)
+				);
+	}
+
+	public static function endpoint_field( $args ) {
+		printf( '<label for="%1$s"><input id="%1$s" name="%1$s" type="url" value="%2$s" placeholder="%3$s" />', esc_attr( $args['label_for'] ), esc_url( get_option( $args['label_for'], $args['default'] ) ), esc_url( $args['default'] ) );
+	}
+
+	public static function auth_settings() {
+		esc_html_e( 'Please specify a remote indieauth authorization and token endpoint.', 'indieauth' );
+	}
+
+	public function get_authorization_endpoint() {
+		$return = get_option( 'indieauth_authorization_endpoint', 'https://indieauth.com/auth' );
+		// Sanity Check
+		if ( empty( $return ) ) {
+			delete_option( 'indieauth_authorization_endpoint' );
+			$return = 'https://indieauth.com/auth';
+		}
+		return $return;
+	}
+
+	public function get_token_endpoint() {
+		$return = get_option( 'indieauth_token_endpoint', 'https://tokens.indieauth.com/token' );
+		// Sanity Check
+		if ( empty( $return ) ) {
+			delete_option( 'indieauth_token_endpoint' );
+			$return = 'https://tokens.indieauth.com/token';
+		}
+		return $return;
+	}
+
+	public function verify_access_token( $token ) {
+		$resp = wp_remote_get(
+			$this->get_token_endpoint(),
+			array(
+				'headers' => array(
+					'Accept'        => 'application/json',
+					'Authorization' => 'Bearer ' . $token,
+				),
+			)
+		);
+		if ( is_wp_error( $resp ) ) {
+			return $resp;
+		}
+
+		$code   = (int) wp_remote_retrieve_response_code( $resp );
+		$body   = wp_remote_retrieve_body( $resp );
+		$params = json_decode( $body, true );
+
+		if ( ( $code / 100 ) !== 2 ) {
+			return new WP_OAuth_Response( 'invalid_request', 'invalid access token', 403 );
+		}
+
+		// look for a user with the same url as the token's `me` value.
+		$user = get_user_by_identifier( $params['me'] );
+		if ( $user instanceof WP_User ) {
+			$params['user'] = $user->ID;
+		} else {
+			return new WP_OAuth_Response( 'unauthorized', __( 'Unable to Find User', 'indieauth' ), 401 );
+		}
+		return $params;
+	}
+
+	public static function verify_authorization_code( $code ) {
+		$args  = array(
+			'headers' => array(
+				'Accept' => 'application/json',
+			),
+		);
+		$query = build_query(
+			array(
+				'code'         => rawurlencode( $code ),
+				'redirect_uri' => wp_login_url( $redirect_uri ),
+				'client_id'    => home_url(),
+			)
+		);
+		$resp  = wp_safe_remote_post( static::get_authorization_endpoint() . '?' . $query, $args );
+		if ( is_wp_error( $resp ) ) {
+			return wp_error_to_oauth_response( $resp );
+		}
+		$code   = wp_remote_retrieve_response_code( $resp );
+		$body   = wp_remote_retrieve_body( $resp );
+		$params = json_decode( $response, true );
```

Comment:
`$params` is never used?

Reply:
I see the mistake. Fixing now.

Reply:
After I tested it, I noticed one function used params, the other reused body. I matched them, but made a mistake. 

I have more PRs before I propose releasing a new version. I want to do something with scopes. I also want some feedback on the settings text.